Parse/Infrastructure/Control/ParseAddUniqueOperation.cs

```diff
@@ -7,61 +7,125 @@
-                return Data.ToList();
+                result[result.IndexOf(matched)] = target;
             }
+            else
+            {
+                result.Add(target);
+            }
+        }
 
-            List<object> result = Conversion.To<IList<object>>(oldValue).ToList();
-            IEqualityComparer<object> comparer = ParseFieldOperations.ParseObjectComparer;
+        return result;
+    }
+
+    public IDictionary<string, object> ConvertToJSON(IServiceHub serviceHub = default)
+    {
+        // Converts the data into JSON-compatible structures
+        var encodedObjects = Data.Select(EncodeForParse).ToList();
 
-            foreach (object target in Data)
+        return new Dictionary<string, object>
+        {
+            ["__op"] = "AddUnique", // Parse operation type
+            ["objects"] = encodedObjects
+        };
+    }
+
+    // Helper method for encoding individual objects
+    private object EncodeForParse(object obj)
+    {
+        return obj switch
+        {
+            // Handle pointers
+            ParseObject parseObj => new Dictionary<string, object>
             {
-                if (target is ParseObject)
-                {
-                    if (!(result.FirstOrDefault(reference => comparer.Equals(target, reference)) is { } matched))
-                    {
-                        result.Add(target);
-                    }
-                    else
-                    {
-                        result[result.IndexOf(matched)] = target;
-                    }
-                }
-                else if (!result.Contains(target, comparer))
-                {
-                    result.Add(target);
-                }
-            }
+                ["__type"] = "Pointer",
+                ["className"] = parseObj.ClassName,
+                ["objectId"] = parseObj.ObjectId
+            },
 
-            return result;
-        }
+            // Handle GeoPoints
+            ParseGeoPoint geoPoint => new Dictionary<string, object>
+            {
+                ["__type"] = "GeoPoint",
+                ["latitude"] = geoPoint.Latitude,
+                ["longitude"] = geoPoint.Longitude
+            },
 
-        public IEnumerable<object> Objects => Data;
+            // Handle Files
+            ParseFile file => new Dictionary<string, object>
+            {
+                ["__type"] = "File",
+                ["name"] = file.Name,
+                ["url"] = file.Url
+            },
+
+            // Handle Relations
+            ParseRelationBase relation => new Dictionary<string, object>
+            {
+                ["__type"] = "Relation",
+                ["className"] = relation.TargetClassName
+            },
+
+            // Handle primitive types
+            string or int or long or float or double or decimal or bool => obj,
+
+            // Handle Bytes
+            byte[] bytes => new Dictionary<string, object>
+            {
+                ["__type"] = "Bytes",
```

Comment:
Should this be separated out/reused

Reply:
Would appreciate what would be better meant by "Should this be separated out/reused".
Do I not handle them explicitly? move to a new file?  where?
Do I would default to removing them from he file first?
Thanks.

Reply:
No worries, my suggestion is whether the `EncodeForParse` method should be moved to a class + new file. If this logic is duplicative elsewhere, we can centralize it

Reply:
Ah, I see now.
Well I would not have much of a say as I believe I followed the existing structure so if the method is there it means I had to include it perhaps to make it complete/work.

Personally, I'd suggest keeping, as It is ONLY used by this file.

Reply:
We could find a why to centralize them though!

But I would let anyone else with the skills and time to help us with that (unless I find the time once again).

I would kindly remind that I focused more on bringing Parity and functionality + security.
I believed and still believe that code clean up could be well done with someone having THAT in mind specifically and clear directives.

Reply:
No worries, happy to leave as is for now if the logic is not duplicative